lib/src/codepage.dart

```diff
@@ -277,6 +277,31 @@ CodePageDecoder _createDecoder(String characters) {
   return _NonBmpCodePageDecoder._(result);
 }
 
+/// An input [ByteConversionSink] for decoders where each input byte can be be
+/// considered independantly.
+class _CodePageDecoderSink implements ByteConversionSink {
+  final Sink<String> _output;
+  final String Function(List<int> input) _convert;
+
+  _CodePageDecoderSink(this._output, this._convert);
+
+  @override
+  void add(List<int> chunk) {
+    _output.add(_convert(chunk));
+  }
+
+  @override
+  void addSlice(List<int> chunk, int start, int end, bool isLast) {
+    add(chunk.sublist(start, end));
```

Comment:
This is the implementation you'd inherit if you extend `ByteConversionSink`, so just do that.

We should then (in a follow-up CL) give the decoder a helper method, or arguments to `convert` itself, so that it can work on a slice of a list, instead of only all of it, so this operation can do its work without allocating a new sublist.

Reply:
Done